components/director/internal/domain/formation/service_test.go

```diff
@@ -824,37 +974,127 @@ func TestServiceUnassignFormation(t *testing.T) {
 			UIDServiceFn: frmtest.UnusedUUIDService(),
 			LabelServiceFn: func() *automock.LabelService {
 				labelService := &automock.LabelService{}
-				labelService.On("GetLabel", ctx, tnt, lblInput).Return(lblSingleFormation, nil)
+				labelService.On("GetLabel", ctx, tnt, applicationLblInput).Return(applicationLblSingleFormation, nil)
 				return labelService
 			},
 			LabelRepoFn: func() *automock.LabelRepository {
 				labelRepo := &automock.LabelRepository{}
 				labelRepo.On("Delete", ctx, tnt, model.ApplicationLabelableObject, objectID, model.ScenariosKey).Return(nil)
 				return labelRepo
 			},
-			AsaServiceFN: func() *automock.AutomaticFormationAssignmentService {
-				return &automock.AutomaticFormationAssignmentService{}
-			},
+			AsaServiceFN:       formation.UnusedASAService,
+			EngineFn:           formation.UnusedEngine,
 			ObjectType:         graphql.FormationObjectTypeApplication,
 			InputFormation:     in,
 			ExpectedFormation:  expected,
 			ExpectedErrMessage: "",
 		},
 		{
-			Name:         "success for tenant",
+			Name:         "success for runtime",
 			UIDServiceFn: frmtest.UnusedUUIDService(),
 			LabelServiceFn: func() *automock.LabelService {
-				return &automock.LabelService{}
+				labelService := &automock.LabelService{}
+				labelService.On("GetLabel", ctx, tnt, runtimeLblInput).Return(runtimeLbl, nil)
+				labelService.On("UpdateLabel", ctx, tnt, runtimeLbl.ID, &model.LabelInput{
+					Key:        model.ScenariosKey,
+					Value:      []string{secondTestFormation},
+					ObjectID:   objectID,
+					ObjectType: model.RuntimeLabelableObject,
+					Version:    0,
+				}).Return(nil)
+				return labelService
+			},
+			LabelRepoFn:  formation.UnusedLabelRepo,
+			AsaServiceFN: formation.UnusedASAService,
+			EngineFn: func() *automock.ScenarioAssignmentEngine {
+				engine := &automock.ScenarioAssignmentEngine{}
+				engine.On("GetScenariosFromMatchingASAs", ctx, objectID).Return(nil, nil)
+				return engine
+			},
+			ObjectType:         graphql.FormationObjectTypeRuntime,
+			InputFormation:     in,
+			ExpectedFormation:  expected,
+			ExpectedErrMessage: "",
+		},
+		{
+			Name:           "success for runtime when formation is coming from ASA",
+			UIDServiceFn:   frmtest.UnusedUUIDService(),
+			LabelServiceFn: formation.UnusedLabelService,
+			LabelRepoFn:    formation.UnusedLabelRepo,
+			AsaServiceFN:   formation.UnusedASAService,
+			EngineFn: func() *automock.ScenarioAssignmentEngine {
+				engine := &automock.ScenarioAssignmentEngine{}
+				engine.On("GetScenariosFromMatchingASAs", ctx, objectID).Return([]string{testFormation}, nil)
+				return engine
+			},
+			ObjectType:         graphql.FormationObjectTypeRuntime,
+			InputFormation:     in,
+			ExpectedFormation:  expected,
+			ExpectedErrMessage: "",
+		},
+		{
+			Name:         "success for runtime if formation do not exist",
```

Comment:
This testCase is the same as "success for runtime". Should use different InputFormation that is not part of the assigned formations